File: src/main/java/online/maestoso/soulforged/recipe/RecipeTables.java
package online.maestoso.soulforged.recipe;

import org.apache.commons.lang3.tuple.Triple;

import java.util.HashMap;

public class RecipeTables {
    public static final HashMap<String, Triple<String, String, String>> TOOL_RECIPES = new HashMap<>();
    public static void initRecipeTables() {
        TOOL_RECIPES.put("soulforged:shortsword", Triple.of("soulforged:shortsword_blade", "soulforged:hilt", "soulforged:handle"));
        TOOL_RECIPES.put("soulforged:broadsword", Triple.of("soulforged:broadsword_blade", "soulforged:wide_hilt", "soulforged:handle"));
        TOOL_RECIPES.put("soulforged:longsword", Triple.of("soulforged:longsword_blade", "soulforged:wide_hilt", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:greatsword", Triple.of("soulforged:greatsword_blade", "soulforged:wide_hilt", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:rapier", Triple.of("soulforged:rapier_blade", "soulforged:slim_hilt", "soulforged:handle"));
        TOOL_RECIPES.put("soulforged:shortspear", Triple.of("soulforged:spearhead", "soulforged:binding", "soulforged:long_shaft"));
        TOOL_RECIPES.put("soulforged:longspear", Triple.of("soulforged:spearhead", "soulforged:binding", "soulforged:very_long_shaft'"));
        TOOL_RECIPES.put("soulforged:javelin", Triple.of("soulforged:spearhead", "soulforged:binding", "soulforged:javelin_shaft"));
        TOOL_RECIPES.put("soulforged:mace", Triple.of("soulforged:macehead", "soulforged:binding", "soulforged:short_shaft"));
        TOOL_RECIPES.put("soulforged:morningstar", Triple.of("soulforged:morningstar_head", "soulforged:binding", "soulforged:short_shaft"));
        TOOL_RECIPES.put("soulforged:greataxe", Triple.of("soulforged:greataxe_head", "soulforged:tough_binding", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:greathammer", Triple.of("soulforged:greathammer_head", "soulforged:tough_binding", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:warhammer", Triple.of("soulforged:warhammer_head", "soulforged:binding", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:axe", Triple.of("soulforged:axe_head", "soulforged:binding", "soulforged:handle"));
        TOOL_RECIPES.put("soulforged:pickaxe", Triple.of("soulforged:pickaxe_head", "soulforged:binding", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:shovel", Triple.of("soulforged:shovel_head", "soulforged:binding", "soulforged:long_handle"));
        TOOL_RECIPES.put("soulforged:hoe", Triple.of("soulforged:hoe_head", "soulforged:binding", "soulforged:long_handle"));
    }
}

File: src/main/kotlin/studio/soulforged/soulforged/recipe/RecipeTables.kt
package studio.soulforged.soulforged.recipe

import net.minecraft.util.Identifier
import java.util.HashMap
import org.apache.commons.lang3.tuple.Triple

object RecipeTables {
    val TOOL_RECIPES = HashMap<String, Triple<Identifier, Identifier, Identifier>>()
    fun initRecipeTables() {
        TOOL_RECIPES["soulforged:shortsword"] = Triple.of(
            Identifier("soulforged:shortsword_blade"),
            Identifier("soulforged:hilt"),
            Identifier("soulforged:handle")
        )
        TOOL_RECIPES["soulforged:broadsword"] = Triple.of(
            Identifier("soulforged:broadsword_blade"),
            Identifier("soulforged:wide_hilt"),
            Identifier( "soulforged:handle")
        )
        TOOL_RECIPES["soulforged:longsword"] = Triple.of(
            Identifier(  "soulforged:longsword_blade"),
            Identifier(  "soulforged:wide_hilt"),
            Identifier(  "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:greatsword"] = Triple.of(
            Identifier( "soulforged:greatsword_blade"),
            Identifier( "soulforged:wide_hilt"),
            Identifier(  "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:rapier"] = Triple.of(
            Identifier(   "soulforged:rapier_blade"),
            Identifier(   "soulforged:slim_hilt"),
            Identifier(   "soulforged:handle")
        )
        TOOL_RECIPES["soulforged:shortspear"] = Triple.of(
            Identifier(    "soulforged:spearhead"),
            Identifier(   "soulforged:binding"),
            Identifier(   "soulforged:long_shaft")
        )
        TOOL_RECIPES["soulforged:longspear"] = Triple.of(
            Identifier("soulforged:spearhead"),
            Identifier("soulforged:binding"),
            Identifier("soulforged:very_long_shaft")
        )
        TOOL_RECIPES["soulforged:javelin"] = Triple.of(
            Identifier("soulforged:spearhead"),
            Identifier("soulforged:binding"),
            Identifier("soulforged:javelin_shaft")
        )
        TOOL_RECIPES["soulforged:mace"] = Triple.of(
            Identifier(    "soulforged:macehead"),
            Identifier(   "soulforged:binding"),
            Identifier(   "soulforged:short_shaft")
        )
        TOOL_RECIPES["soulforged:morningstar"] = Triple.of(
            Identifier(  "soulforged:morningstar_head"),
            Identifier(  "soulforged:binding"),
            Identifier(  "soulforged:short_shaft")
        )
        TOOL_RECIPES["soulforged:greataxe"] = Triple.of(
            Identifier(  "soulforged:greataxe_head"),
            Identifier(  "soulforged:tough_binding"),
            Identifier(  "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:greathammer"] = Triple.of(
            Identifier( "soulforged:greathammer_head"),
            Identifier(  "soulforged:tough_binding"),
            Identifier(  "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:warhammer"] = Triple.of(
            Identifier( "soulforged:warhammer_head"),
            Identifier( "soulforged:binding"),
            Identifier( "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:axe"] = Triple.of(
            Identifier("soulforged:axe_head"),
            Identifier("soulforged:binding"),
            Identifier("soulforged:handle")
        )
        TOOL_RECIPES["soulforged:pickaxe"] = Triple.of(
            Identifier(  "soulforged:pickaxe_head"),
            Identifier(  "soulforged:binding"),
            Identifier(  "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:shovel"] = Triple.of(
            Identifier( "soulforged:shovel_head"),
            Identifier(  "soulforged:binding"),
            Identifier(  "soulforged:long_handle")
        )
        TOOL_RECIPES["soulforged:hoe"] = Triple.of(
            Identifier( "soulforged:hoe_head"),
            Identifier( "soulforged:binding"),
            Identifier( "soulforged:long_handle")
        )
    }
}